### ISSUE TITLE
HDD/SSD Configuration

### ISSUE THREAD

**Issue by JohnOH** [jeelabs/swipe-dev#5]
Since we plan to replicate some cloud service shall we decide to had a minimal SDcard bootstrapping an external SSD?

**Comment by jcw** [comment]
Yes, sounds like a good idea. Using SD cards for storage is a time-bomb. I will need to be an option though, if the only use is for PiHole. I think we'll need to define what the minimal setup will be.

**Comment by jcw** [comment]
DietPi supports moving root to an external drive, leaving the SD to boot and then mount `/` that way.

The other option is the Pi 3B+'s option to change booting directly from USB, but that's irreversible, and very Pi-specific - I suspect that's why DietPi offers the other mechanism. I assume the Pi way means no SD card is needed, whereas DietPi's approach needs a minimal SD during the boot step.

I'd be inclined to stick to DietPi's approach. Hve used it in the past, it definitely works. But before we start going there, I think we first need to identify the first goal which will need the extra disk space (and will lead to the higher disk wear and risk of failure).

This probably belongs in separate issues, but some use cases I'd be interested in would be file/photo storage and sharing, combined with backup to another Swipe unit (based on either rsync or duplicacy).

**Comment by jcw** [comment]
Another idea would be to use something like the ODROID-HC1 or HC2 for such a setup. It has no WiFi, and will need an extra dongle. The HC1 (2.5" 5V) and HC2 (3.5" 12V) are both supported by DietPi.

**Comment by jcw** [comment]
I've attached this to the "Postponed" milestone, because the first iterations of Swipe are now switching to an asymmetric model: a "thin" Swipe Node which is essentially just a pass-through, and a "fat" Swipe server, where all the data lives, for access by the nodes.

At some point we may want to expand the thin nodes, but for now I'd like to focus on the above model. The Swipe Server can, and probably _should_, be fairly beefy, for good responsiveness to node accesses. Also, initially at least, it'll be all about Nextcloud (which is designed to scale well).
